projects/components/src/filter-bar/filter-bar.service.ts

```diff
@@ -1,17 +1,27 @@
 import { Injectable } from '@angular/core';
 import { NavigationService } from '@hypertrace/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 import { FilterAttribute } from './filter-attribute';
 import { Filter } from './filter/filter-api';
 import { FilterParserService } from './filter/parser/filter-parser.service';
 
+export type GetFiltersFunction = (attributes: FilterAttribute[]) => Filter[];
+
 @Injectable({ providedIn: 'root' })
 export class FilterBarService {
   private static readonly FILTER_QUERY_PARAM: string = 'filter';
 
+  public readonly urlFilterChanges$: Observable<GetFiltersFunction>;
```

Comment:
IMO it would be cleaner to have a function taking an array of attributes and returning an observable<filter[]>. Thoughts?

Reply:
At first, you convinced me to leave it as is, then I dug in - the current implementation doesn't handle the case you described either, because the function is only invoked if `urlFilterChanges$` emits. So we'd need to do something like converting the input to an observable too and zipping them together - at which point, there's no extra management of subscriptions between the two approaches.

Reply:
So close! 

But really, good catch! Updating.